### PR TITLE
Make sure we don't hide the cursor until the IME starts

### PR DESCRIPTION
**Pull request by DHowett**
Some IME implementations do not produce composition strings, and their
users have come to rely on the cursor that conhost traditionally left on
until a composition string showed up. We shouldn't hide the cursor until
we get a string (as opposed to hiding it when composition begins) so as
to not break those IMEs.

Related to #6207.

Fixes MSFT:29219348

### CODE REVIEW

**Comment by DHowett** [comment]
/cc @j4james

I had to revert a little bit of your niceness from #6207. Not much of it!

**Review by miniksa — APPROVED** [review]

File: src/host/convarea.cpp [hunk 1]
```diff
@@ -104,8 +104,16 @@ void WriteConvRegionToScreen(const SCREEN_INFORMATION& ScreenInfo,
     gci.LockConsole();
     auto unlock = wil::scope_exit([&] { gci.UnlockConsole(); });
 
-    ConsoleImeInfo* const pIme = &gci.ConsoleIme;
-    pIme->SaveCursorVisibility();
+    // MSFT:29219348 Some IME implementations do not
+    // produce composition strings, and their users
+    // have come to rely on the cursor that conhost
+    // traditionally left on until a composition
+    // string showed up.
```
Comment (DHowett) [inline src/host/convarea.cpp:111]:
```suggestion
    // string showed up.
    // One such IME is WNWB's "Universal Wubi input
    // method" from wnwb.com (version 10+).
```

**Review by carlos-zamora — APPROVED** [review]

File: src/host/convarea.cpp [hunk 2]
```diff
@@ -104,8 +104,18 @@ void WriteConvRegionToScreen(const SCREEN_INFORMATION& ScreenInfo,
...
+    // have come to rely on the cursor that conhost
+    // traditionally left on until a composition
+    // string showed up.
+    // One such IME is WNWB's "Universal Wubi input
+    // method" from wnwb.com (version 10+).
+    // We shouldn't hide the cursor here so as to not
+    // break those IMEs.
+    //
+    // ConsoleImeInfo* const pIme = &gci.ConsoleIme;
+    // pIme->SaveCursorVisibility();
```
Comment (DHowett) [inline src/host/convarea.cpp:118]:
```suggestion
    // MSFT:29219348 Some IME implementations do not produce composition strings, and
    // their users have come to rely on the cursor that conhost traditionally left on
    // until a composition string showed up.
    // One such IME is WNWB's "Universal Wubi input method" from wnwb.com (v. 10+).
    // We shouldn't hide the cursor here so as to not break those IMEs.
```

**Comment by DHowett** [comment]
@msftbot merge this in 3 minutes

**Comment by j4james** [comment]
> I had to revert a little bit of your niceness from #6207. Not much of it!

My own fault for being overzealous with the refactoring. Sorry for breaking things.

**Comment by DHowett** [comment]
Never a problem! You can't make an omelet without breaking a few lions.

**Pull request merged**